sway/config/output.c

```diff
@@ -500,25 +480,17 @@ static void queue_output_config(struct output_config *oc,
 		sway_log(SWAY_DEBUG, "Set %s adaptive sync to %d", wlr_output->name,
 			oc->adaptive_sync);
 		wlr_output_state_set_adaptive_sync_enabled(pending, oc->adaptive_sync == 1);
-		if (oc->adaptive_sync == 1 && !wlr_output_test_state(wlr_output, pending)) {
-			sway_log(SWAY_DEBUG, "Adaptive sync failed, ignoring");
-			wlr_output_state_set_adaptive_sync_enabled(pending, false);
-		}
 	}
 
 	if (oc && oc->render_bit_depth != RENDER_BIT_DEPTH_DEFAULT) {
-		const uint32_t *fmts = bit_depth_preferences[oc->render_bit_depth];
-		assert(fmts);
-
-		for (size_t i = 0; fmts[i] != DRM_FORMAT_INVALID; i++) {
-			wlr_output_state_set_render_format(pending, fmts[i]);
-			if (wlr_output_test_state(wlr_output, pending)) {
-				break;
-			}
-
-			sway_log(SWAY_DEBUG, "Preferred output format 0x%08x "
-				"failed to work, falling back to next in "
-				"list, 0x%08x", fmts[i], fmts[i + 1]);
+		if (oc->render_bit_depth == RENDER_BIT_DEPTH_10 &&
+			render_format_is_10bit(output->wlr_output->render_format)) {
+			// 10-bit was set successfully before, try to save some tests by reusing the format
+			wlr_output_state_set_render_format(pending, output->wlr_output->render_format);
+		} else if (oc->render_bit_depth == RENDER_BIT_DEPTH_10) {
+			wlr_output_state_set_render_format(pending, DRM_FORMAT_XRGB2101010);
+		} else {
+			wlr_output_state_set_render_format(pending, DRM_FORMAT_XRGB8888);
```

Comment:
This is an explicit request for an 8-bit format, so there is no fallback in effect.

(In the last commit, the logs also get refactored slightly to just be the dump_output_state calls, rather than individual "trying xyz" logs)